pkg/controllers/jobflow/jobflow_controller_action.go

```diff
@@ -285,25 +273,32 @@ func (jf *jobflowcontroller) loadJobTemplateAndSetJob(jobFlow *v1alpha1flow.JobF
 }
 
 func (jf *jobflowcontroller) deleteAllJobsCreatedByJobFlow(jobFlow *v1alpha1flow.JobFlow) error {
-	selector := labels.NewSelector()
-	jobList, err := jf.jobLister.Jobs(jobFlow.Namespace).List(selector)
+	jobList, err := jf.getAllJobsCreatedByJobFlow(jobFlow)
 	if err != nil {
 		return err
 	}
 
 	for _, job := range jobList {
-		if len(job.OwnerReferences) > 0 {
-			for _, reference := range job.OwnerReferences {
-				if reference.Kind == helpers.JobFlowKind.Kind && reference.Name == jobFlow.Name {
-					err := jf.vcClient.BatchV1alpha1().Jobs(jobFlow.Namespace).Delete(context.Background(), job.Name, metav1.DeleteOptions{})
-					if err != nil {
-						klog.Errorf("Failed to delete job of JobFlow %v/%v: %v",
-							jobFlow.Namespace, jobFlow.Name, err)
-						return err
-					}
-				}
-			}
+		err := jf.vcClient.BatchV1alpha1().Jobs(jobFlow.Namespace).Delete(context.Background(), job.Name, metav1.DeleteOptions{})
+		if err != nil {
+			klog.Errorf("Failed to delete job of JobFlow %v/%v: %v",
+				jobFlow.Namespace, jobFlow.Name, err)
+			return err
 		}
 	}
 	return nil
 }
+
+func (jf *jobflowcontroller) getAllJobsCreatedByJobFlow(jobFlow *v1alpha1flow.JobFlow) ([]*v1alpha1.Job, error) {
+	var flowNames []string
+	for _, flow := range jobFlow.Spec.Flows {
+		flowNames = append(flowNames, GetTemplateString(jobFlow.Namespace, flow.Name))
+	}
+	selector := labels.NewSelector()
+	r, err := labels.NewRequirement(CreatedByJobTemplate, selection.In, flowNames)
```

Comment:
nit: I think you need to update the comment of `CreatedByJobTemplate`, it specifies it is an annotation there